File: java/template-renderer/src/main/java/com/rbleggi/templaterenderer/renderers/PDFRenderer.java
package com.rbleggi.templaterenderer.renderers;

import com.itextpdf.kernel.pdf.PdfDocument;
import com.itextpdf.kernel.pdf.PdfWriter;
import com.itextpdf.layout.Document;
import com.itextpdf.layout.element.Paragraph;
import com.rbleggi.templaterenderer.core.TemplateRenderer;

import java.io.ByteArrayOutputStream;
import java.util.Map;

public class PDFRenderer extends TemplateRenderer {
    @Override
    public byte[] render(Map<String, String> data) {
        var outputStream = new ByteArrayOutputStream();
        var document = new Document(new PdfDocument(new PdfWriter(outputStream)));

        System.out.println("Generating PDF Template...");

        document.add(new Paragraph("Title: " + data.get("title")));
        document.add(new Paragraph("Content: " + data.get("content")));

        document.close();

        System.out.println("PDF Template Generated Successfully!");

        return outputStream.toByteArray();
    }
}

File: kotlin/template-renderer/src/main/kotlin/com/rbleggi/templaterenderer/renderers/PDFRenderer.kt
package com.rbleggi.templaterenderer.renderers

import com.itextpdf.kernel.pdf.PdfDocument
import com.itextpdf.kernel.pdf.PdfWriter
import com.itextpdf.layout.Document
import com.itextpdf.layout.element.Paragraph
import com.rbleggi.templaterenderer.core.TemplateRenderer
import java.io.ByteArrayOutputStream

class PDFRenderer : TemplateRenderer() {
    override fun render(data: Map<String, String>): ByteArray {
        val outputStream = ByteArrayOutputStream()
        val document = Document(PdfDocument(PdfWriter(outputStream)))

        println("Generating PDF Template...")

        document.add(Paragraph("Title: ${data["title"]}"))
        document.add(Paragraph("Content: ${data["content"]}"))

        document.close()

        println("PDF Template Generated Successfully!")

        return outputStream.toByteArray()
    }
}